src/webgpu/api/validation/encoding/encoder_state.spec.ts

```diff
@@ -19,4 +19,67 @@ TODO:
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { ValidationTest } from '../validation_test.js';
 
-export const g = makeTestGroup(ValidationTest);
+class F extends ValidationTest {
+  beginRenderPass(commandEncoder: GPUCommandEncoder): GPURenderPassEncoder {
+    const attachmentTexture = this.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 16, height: 16, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+
+    return commandEncoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: attachmentTexture.createView(),
+          clearValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+          loadOp: 'clear',
+          storeOp: 'store',
+        },
+      ],
+    });
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('pass_end_invalid_order')
+  .desc(
+    `
+  Test that beginning a [compute|render] pass before ending the previous pass causes an error.
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine(`IsRenderPassFirst`, [true, false])
+      .combine(`IsComputePassSecond`, [true, false])
```

Comment:
nit: This naming is a little confusing, both the wording and why "true" means render for the first pass, but it means compute for the second pass. How about something like this?
```
      .combine('pass0Type', ['compute', 'render'])
      .combine('pass1Type', ['compute', 'render'])
```

Reply:
ok, it seems more readable. Fixed.